libraries/common/bytearray.effekt

```diff
@@ -12,32 +12,47 @@ extern type ByteArray
 extern global def allocate(size: Int): ByteArray =
   js "(new Uint8Array(${size}))"
   llvm """
-    %arr = call %Pos @c_bytearray_new(%Int ${size})
-    ret %Pos %arr
+    %object_size = add i64 ${size}, 16
+    %object_alloc = tail call noalias ptr @malloc(i64 noundef %object_size)
+    store i64 0, ptr %object_alloc, align 8
+    %object_data_ptr = getelementptr inbounds i8, ptr %object_alloc, i64 8
+    store ptr @c_bytearray_erase_noop, ptr %object_data_ptr, align 8
```

Comment:
Define this `@c_bytearray_erase_noop` here in Effekt instead of in C (In analogy to the eraser for arrays).